docs/architecture/adr-005-battle-portal-ui.html

```
@@ -0,0 +1,701 @@
+<li><code>health-check-handler</code> で <code>lastResult=fail</code> 遷移時 (= 直前 tick が ok だった場合) に attack-detected event を 1 件書く。<b>連続失敗 tick では重複 write しない</b> (= row 爆発防止の hard guard)</li>
+<li><code>lookup.ts</code> の <code>endpointsHealth</code> を <code>aggregate</code> (= ok count / total / overallStatus) に絞った view を新設、participant API でこちらを返す</li>
+<li><code>GET /portal/me/battle-attacks?jobId=&amp;sinceMin=</code> Lambda route + ParticipantPortalLambda の IAM に必要なら Query 権限を追加</li>
+<li><b><code>sinceMin</code> の上限は 60 分</b>とし、defaults は 30 分。Battle UI の uptime ％は「直近 5 分」、attack history は「直近 60 分」を最大として固定 (= read RCU 暴発の防止策)</li>
+</ol>
+
+<h4>追加するテスト</h4>
+<ul>
+<li><code>health-check-handler</code> が <code>ok → fail</code> 遷移時に attack-detected event を <b>1 件のみ</b> 書く (連続 fail tick で重複しない)</li>
+<li><code>lookup.ts</code> の aggregate view に <code>endpointsHealth</code> 内の URL / endpoint 名が含まれない (snapshot test)</li>
+<li><code>GET /portal/me/battle-attacks</code> が <code>jobId</code> / <code>sinceMin</code> で正しい <code>SCORE#&lt;eventId&gt;#&lt;jobId&gt;</code> を Query (mock DDB)</li>
+<li><code>sinceMin</code> > 60 を渡すと 400 (= clamp ではなく reject、operator が UI で渡す前提)</li>
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)